DragaliaAPI/DragaliaAPI/Features/StorySkip/StorySkipService.cs

```diff
@@ -0,0 +1,191 @@
+                await rewardService.GrantReward(new Entity(entity, id, quantity));
+                logger.LogDebug("Completed quest mission {missionId}", i);
+                clearSet.Add(new AtgenMissionsClearSet(id, entity, quantity, i + 1));
+            }
+        }
+
+        List<AtgenFirstClearSet> completeSet = new();
+
+        if (currentState.Any(x => !x) && newState.All(x => x))
+        {
+            await rewardService.GrantReward(
+                new Entity(
+                    rewardData.MissionCompleteEntityType,
+                    rewardData.MissionCompleteEntityId,
+                    rewardData.MissionCompleteEntityQuantity
+                )
+            );
+            logger.LogDebug("Granting bonus for completing all missions");
+            completeSet.Add(
+                new AtgenFirstClearSet(
+                    rewardData.MissionCompleteEntityId,
+                    rewardData.MissionCompleteEntityType,
+                    rewardData.MissionCompleteEntityQuantity
+                )
+            );
+        }
+
+        return new QuestMissionStatus(newState, clearSet, completeSet);
+    }
+
+    public async Task IncreaseFortLevels()
+    {
+        Dictionary<FortPlants, FortConfig> fortConfigs = FortConfigurations.FortConfigs;
+
+        foreach (KeyValuePair<FortPlants, FortConfig> keyValuePair in fortConfigs)
+        {
+            FortPlants fortPlant = keyValuePair.Key;
+            FortConfig fortConfig = keyValuePair.Value;
+            await fortRepository.AddToStorage(
+                fortPlant,
+                fortConfig.BuildCount,
+                true,
+                fortConfig.Level
+            );
+        }
+
+        IEnumerable<BuildList> buildListEnum = await fortService.GetBuildList();
+        foreach (BuildList buildList in buildListEnum)
+        {
+            if (fortConfigs.TryGetValue(buildList.PlantId, out FortConfig fortConfig))
+            {
+                DbFortBuild build = await fortRepository.GetBuilding((long)buildList.BuildId);
+                if (build.Level < fortConfig.Level)
+                {
+                    build.Level = fortConfig.Level;
+                    build.BuildStartDate = DateTimeOffset.UnixEpoch;
+                    build.BuildEndDate = DateTimeOffset.UnixEpoch;
+                }
+            }
+        }
+    }
+
+    public async Task IncreasePlayerLevel()
+    {
+        const int MaxLevel = 60;
+        const int MaxExp = 69990;
+        DbPlayerUserData data = await userDataRepository.GetUserDataAsync();
+        data.TutorialFlag = 16640603;
+        data.TutorialStatus = 60999;
+        data.StaminaSingle = 999;
+        data.StaminaMulti = 99;
+
+        if (data.Exp < MaxExp)
+        {
+            data.Exp = MaxExp;
+        }
+
+        if (data.Level < MaxLevel)
+        {
+            data.Level = MaxLevel;
+        }
+    }
+
+    public async Task OpenTreasure(int questTreasureId, CancellationToken cancellationToken)
```

Comment:
Do we have to open treasure as part of the story skip? The wiki doesn't mention it anywhere

Reply:
This was one of those nuances that I was unsure about. I can remove the treasure opening.